Lib/threading.py

```diff
@@ -1342,6 +1342,12 @@ def run(self):
             self.function(*self.args, **self.kwargs)
         self.finished.set()
 
+    def __enter__(self):
+        self.start()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.cancel()
```

Comment:
this should probably join the Timer thread

Reply:
Why block the main thread for a canceled timer?

Reply:
The main thread won't be blocked very long, it's to support use cases like: 

```python
with Timer(1, print, ("Trust me, I'm on it...",)):
    time.sleep(2)
os.fork()
```